### DISCUSSION TITLE
Payment types & currencies

### DISCUSSION THREAD

**Discussion by valentinhttr** [inisos/bank4j#6] [Q&A]
Hi,

According to this document : 
[ig-credit-transfer-v2.0.2-en.pdf](https://github.com/inisos/bank4j/files/13120752/ig-credit-transfer-v2.0.2-en.pdf)

I'm wondering how to make payments in different currencies (other than SEPA) knowing that I have one bank account per currency:
- CHF
- EUR
- USD
- GBP

At first I thought I'd make one PAIN file per account, but on this image (taken from the pdf) SEPA is only for the currency EUR.
<img width="718" alt="CleanShot 2023-10-24 at 18 20 16@2x" src="https://github.com/inisos/bank4j/assets/20521112/12f52e78-e056-4f4f-953c-a7f2246a458e">

Would it be possible to include national and cross-border payments in this package @pblanchardie please?

If you have any further information, I'd be delighted to hear from you.

Thanks

**Comment by pblanchardie** [comment]
Hi,

Indeed, it has only been used with SEPA.

Yes, you'll need to generate 1 message per debtor account.

Then select the payment type
- D: domestic CHF/EUR
- S: SEPA for cross-border EUR
- X: cross border all other currencies (USD/GBP)

As you've probably noticed, you can generate messages for any service level code (SvcLvl, according to Payments External Code Lists [8]) by using the following:

```java
CreditTransferOperation creditTransfer = Bank.jaxbCreditTransfer()
    .serviceLevelCode("CHBCC")
    // etc.
```
see https://www.iso20022.org/catalogue-messages/additional-content-messages/external-code-sets

But you can't specify the other PaymentTypeInformation (PmtTpInf) elements: InstrPrty, LclInstrm, CtgyPurp. Do you need some?

Regarding Creditor Agent, only BIC is supported but I can easily add support for more FinancialInstitutionIdentification fields. What do you need for domestic CH/LI and cross-border? Could you provide sample data for such identification?

Feel free to request support for additional fields.

**Reply by valentinhttr** [reply]
Thank you for your reply.

This is perfect for the `serviceLevelCode`, I hadn't seen that it could be specified. So I'm thinking of defining the service code as follows: 

- D - domestic: `CHBCC`
- S - SEPA: `SEPA`
- X - cross borders: `URGP`

After reading the above PDF, the `InstrPrty`, `LclInstrm`, `CtgyPurp` fields don't need to be specified.

Regarding Creditor Agent, I have created a feature request #7 .
For many payments, even international, I do not necessarily have an IBAN, even though I know that this is recommended by the ISO20022 standard.
